cpp/src/arrow/util/bit_util.h

```diff
@@ -116,9 +116,11 @@ static constexpr uint8_t kBytePopcount[] = {
 
 // Returns the ceil of value/divisor
 constexpr int64_t CeilDiv(int64_t value, int64_t divisor) {
-  return value / divisor + (value % divisor != 0);
+  return (value - 1) / divisor + 1;
 }
 
+constexpr int64_t CeilDiv8(int64_t value) { return ((value - 1) >> 3) + 1; }
```

Comment:
This shouldn't be needed, the compiler will properly apply the constant propagation optimization since this is a `constexpr` function.

Reply:
@fsaintjacques Thanks a lot for your suggestion.
The additional utility was not intended for performance improvement. It was to make code clearer.

I have removed it in the new commit. Please take another look. Thank you in advance.